### PR TITLE
Add com.visualstudio.code.tool.host-spawn

### PR DESCRIPTION
**Pull request by mdmundo**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** [request to submit to Flathub](https://github.com/1player/host-spawn/issues/24)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: flathub.json
Comment (hfiguiere) [inline flathub.json:1]: you don't need `only-arches`. But you probably need `    "skip-icons-check": true`

**Comment by hfiguiere** [comment]
bot, build com.visualstudio.code.tool.host-spawn

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Missing an appstream file.

**Comment by mdmundo** [comment]
bot, build com.visualstudio.code.tool.host-spawn

**Comment by mdmundo** [comment]
bot, build com.visualstudio.code.tool.host-spawn

**Comment by mdmundo** [comment]
bot, build com.visualstudio.code.tool.host-spawn

File: com.visualstudio.code.tool.host-spawn.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+  <id>com.visualstudio.code.tool.host-spawn</id>
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
```
Comment (hfiguiere) [inline com.visualstudio.code.tool.host-spawn.metainfo.xml:5]:
```suggestion
  <project_license>MIT-0</project_license>
```

according to the repository

**Comment by 1player** [comment]
As the author of host-spawn, LGTM, go ahead :)

**Comment by hfiguiere** [comment]
bot, build com.visualstudio.code.tool.host-spawn

File: com.visualstudio.code.tool.host-spawn.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+  <id>com.visualstudio.code.tool.host-spawn</id>
+  <metadata_license>MIT-0</metadata_license>
```
Comment (hfiguiere) [inline com.visualstudio.code.tool.host-spawn.metainfo.xml:4]: seems that appstream doesn't like this. We prefer `CC0-1.0` for the metadata.

**Comment by mdmundo** [comment]
bot, build com.visualstudio.code.tool.host-spawn

**Comment by mdmundo** [comment]
@hfiguiere it can be checked again

**Comment by AlbertoFabbri93** [comment]
Is this getting merged? I have the problem this merge request is trying to fix...

**Comment by mdmundo** [comment]
It is ready for merging. For now you can try the [build](https://github.com/flathub/flathub/pull/4267#issuecomment-1607162098) command, wait a few minutes and install it after build completion ([like this](https://github.com/flathub/flathub/pull/4267#issuecomment-1607167159)).

**Comment by AlbertoFabbri93** [comment]
Can I install this "branch" side by side with my regular install?

**Comment by mdmundo** [comment]
> Can I install this "branch" side by side with my regular install?

@AlbertoFabbri93 Yeah, but keep in mind it becomes unavailable after some time. If you need to install it later again, then run the command again.

**Comment by AlbertoFabbri93** [comment]
bot, build com.visualstudio.code.tool.host-spawn

**Comment by AlbertoFabbri93** [comment]
I have installed the extension and added the following to my user `settings.json`:
```
"terminal.integrated.defaultProfile.linux": "host-spawn",
    "terminal.integrated.env.linux": {
        "LD_PRELOAD": null,
    },
    "terminal.integrated.profiles.linux": {
        "host-bash": {
            "path": "/usr/bin/flatpak-spawn",
            "args": [ "--host", "--env=TERM=xterm-256color", "bash" ]
        },
        "local": {
            "path": "bash"
        },
        "host-spawn": {
            "path": "/app/bin/host-spawn",
            "args": ["bash"],
            "icon": "terminal-bash",
            "overrideName": true
        },
    },
```
But in the folder `/app/bin/` there is no executable called `host-spawn`. Should I add something to the command line to start vscode with the extension enabled?

**Comment by mdmundo** [comment]
iirc, u don't need the `/app/bin/` prefix

**Comment by AlbertoFabbri93** [comment]
I saw that path in the updated readme file...

Thank you very much, it works! And the errors are gone :)

**Comment by alatiera** [comment]
bot, build com.visualstudio.code.tool.host-spawn

**Comment by alatiera** [comment]
> u probably want to run `which host-spawn` to get the path

fwiw it should be `/app/tools/host-spawn/bin/host-spawn`

Seems to work okay so far. Honestly we should probably make that the default in the vscode faltpak cause both the current bash config and the hacked `script` snippet from https://github.com/flathub/com.visualstudio.code/issues/315 are not working well.

```
        "host-spawn": {
            "path": "/app/tools/host-spawn/bin/host-spawn",
            "args": ["bash"],
            "icon": "terminal-bash",
            "overrideName": true
        },
```

IMO better to go with https://github.com/flathub/com.visualstudio.code/pull/375

**Pull request closed without merging**